src/m.gl.js

```diff
@@ -47,6 +47,11 @@ please.gl.set_context = function (canvas_id, options) {
 
     this.canvas = document.getElementById(canvas_id);
     please.__create_canvas_overlay();
+
+    if (options == '2d') {
+        return;
```

Comment:
I think please.gl.ctx being null should not indicate that we're in 2D mode, but rather something else should indicate that its in 2D mode.  The reason for this, is if someone calls stuff before set_context, then there should be an entirely different behavior (possibly throwing an error) than assuming 2D or 3D mode, because the user never explicitly stated what they needed.